packages/api/src/hooks/useP2PAdvertList.ts

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import useInfiniteQuery from '../useInfiniteQuery';
+import useAuthorize from './useAuthorize';
+
+/**
+ * This custom hook returns available adverts for use with 'p2p_order_create' by calling 'p2p_advert_list' endpoint
+ */
+const useP2PAdvertList = (
+    payload?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_advert_list'>>[1]>['payload'],
+    config?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_advert_list'>>[1]>['options']
```

Comment:
These props are required since most of the attributes are UI dependent, based on what the user selects in the UI: Buy/Sell tab, what advertiser was selected etc